src/managers/KeybindManager.cpp

```diff
@@ -1153,14 +1181,38 @@ void CKeybindManager::moveFocusTo(std::string args) {
     const auto PWINDOWTOCHANGETO = PLASTWINDOW->m_bIsFullscreen ? g_pCompositor->getNextWindowOnWorkspace(PLASTWINDOW, arg == 'u' || arg == 't' || arg == 'r') :
                                                                   g_pCompositor->getWindowInDirection(PLASTWINDOW, arg);
 
+    // Found window in direction, switch to it
     if (PWINDOWTOCHANGETO) {
         switchToWindow(PWINDOWTOCHANGETO);
-    } else {
-        const auto PWINDOWNEXT = g_pCompositor->getNextWindowOnWorkspace(PLASTWINDOW, true);
-        if (PWINDOWNEXT) {
-            switchToWindow(PWINDOWNEXT);
+        return;
+    }
+
+    Debug::log(LOG, "No window found in direction %c, looking for a monitor", arg);
+
+    // No window found in direction, switch to workspace in direction
+    const auto PNEWMONITOR = g_pCompositor->getMonitorInDirection(arg);
+    if (PNEWMONITOR && PNEWMONITOR->activeWorkspace != -1) {
+        const auto PNEWWORKSPACE = g_pCompositor->getWorkspaceByID(PNEWMONITOR->activeWorkspace);
+        if (PNEWWORKSPACE) {
+            g_pCompositor->focusWindow(nullptr);
```

Comment:
what if there is a floating window? You should do the same as above, getlastfocusedwindow